src/components/plots/barPlot.ts

```diff
@@ -301,13 +303,8 @@ export module Plots {
       var valueScale = this._isVertical ? this.y().scale : this.x().scale;
       if (valueScale instanceof QuantitativeScale) {
         var qscale = <QuantitativeScale<any>> valueScale;
-        if (this._baselineValue != null) {
-          qscale.addPaddingException(this, this._baselineValue);
-          qscale.addIncludedValue(this, this._baselineValue);
-        } else {
-          qscale.removePaddingException(this);
-          qscale.removeIncludedValue(this);
-        }
+        qscale.addPaddingExceptionsProvider(this._baselineValueProvider);
+        qscale.addIncludedValuesProvider(this._baselineValueProvider);
```

Comment:
Same complaint here: The logic has changed, because the Providers are never removed.